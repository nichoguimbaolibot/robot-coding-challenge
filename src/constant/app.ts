export const MAX_BOXES = 5;
