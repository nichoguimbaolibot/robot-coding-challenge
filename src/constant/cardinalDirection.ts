export const NORTH = 'North';
export const EAST = 'East';
export const SOUTH = 'South';
export const WEST = 'West';

export const CARDINAL_DIRECTION = [
  {
    id: 1,
    direction: 0,
    position: NORTH,
  },
  {
    id: 2,
    direction: 90,
    position: EAST,
  },
  {
    id: 3,
    direction: 180,
    position: SOUTH,
  },
  {
    id: 4,
    direction: 270,
    position: WEST,
  },
];

export interface ICardinalDirection {
  direction?: number;
  id?: number;
  position?: string;
}
