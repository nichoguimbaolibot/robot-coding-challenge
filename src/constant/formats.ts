export const DATE_FORMAT = 'LTS';
