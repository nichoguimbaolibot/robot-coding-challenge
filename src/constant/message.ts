const MESSAGES = {
  INVALID_POSITION: 'Buddy~ It was an invalid position.',
  MOVEMENT_NOT_ALLOWED: 'Hey!! I will fall off the table!!',
  ROBOT_NOT_FOUND: 'Im not in the table broooo~',
};

export default MESSAGES;
