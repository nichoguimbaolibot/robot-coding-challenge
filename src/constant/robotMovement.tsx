export const ROBOT_MOVEMENT = {
  DEFAULT_X_AXIS_POSITION: 49,
  DEFAULT_Y_AXIS_POSITION: 420,
  DEFAULT_X_AXIS_MOVEMENT: 96,
  DEFAULT_Y_AXIS_MOVEMENT: 103,
};
