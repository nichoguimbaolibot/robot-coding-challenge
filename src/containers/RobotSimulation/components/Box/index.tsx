import React from 'react';

interface IBox {
  rowBoxes: number;
}

const Box: React.FunctionComponent<IBox> = ({ rowBoxes }): JSX.Element => (
  <div data-testid="box-container-test" className="box-container">
    {Array(rowBoxes)
      .fill(Math.random())
      .map((item: number, index: number) => (
        <div data-testid="box-test" className="box" key={index} />
      ))}
  </div>
);

export default Box;
