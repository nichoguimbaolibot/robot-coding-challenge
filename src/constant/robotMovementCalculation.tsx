import { ROBOT_MOVEMENT } from 'constant/robotMovement';

export const robotCalculation = {
  getXAxisMovement: (xAxis: number): number => {
    return (
      xAxis * ROBOT_MOVEMENT.DEFAULT_X_AXIS_MOVEMENT +
      ROBOT_MOVEMENT.DEFAULT_X_AXIS_POSITION
    );
  },
  getYAxisMovement: (yAxis: number): number => {
    return (
      yAxis * ROBOT_MOVEMENT.DEFAULT_X_AXIS_MOVEMENT +
      ROBOT_MOVEMENT.DEFAULT_X_AXIS_POSITION
    );
  },
};
