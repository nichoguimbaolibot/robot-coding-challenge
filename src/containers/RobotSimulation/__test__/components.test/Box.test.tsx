import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Box from 'containers/RobotSimulation/components/Box';

import { MAX_BOXES } from 'constant/app';

afterEach(cleanup);

it('renders BOX component', () => {
  const { asFragment } = render(<Box rowBoxes={MAX_BOXES} />);
  expect(asFragment()).toMatchSnapshot();
});

it('expect it to have a box-container classname', () => {
  const { getByTestId } = render(<Box rowBoxes={MAX_BOXES} />);
  expect(getByTestId('box-container-test')).toHaveClass('box-container');
});
