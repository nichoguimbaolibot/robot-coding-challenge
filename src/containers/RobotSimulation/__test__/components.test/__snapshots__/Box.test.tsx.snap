// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BOX component 1`] = `
<DocumentFragment>
  <div
    class="box-container"
    data-testid="box-container-test"
  >
    <div
      class="box"
      data-testid="box-test"
    />
    <div
      class="box"
      data-testid="box-test"
    />
    <div
      class="box"
      data-testid="box-test"
    />
    <div
      class="box"
      data-testid="box-test"
    />
    <div
      class="box"
      data-testid="box-test"
    />
  </div>
</DocumentFragment>
`;
